### PR TITLE
Remove duplicate DataSegmentTest (fixes #6064)

### LINKED ISSUE #6064: Duplicate DataSegmentTest
Seems that `io.druid.timeline.DataSegmentTest` is duplicated in druid-api and druid-server modules.

### PR DESCRIPTION
**Pull request by jihoonson**
There are two almost identical `DataSegmentTest`s in `druid-core` and `druid-server`. I removed the one in `druid-server` since `DataSegment` is in `druid-core`.

### CODE REVIEW

File: core/src/test/java/org/apache/druid/timeline/DataSegmentTest.java [hunk 1]
```diff
@@ -51,7 +51,7 @@
 public class DataSegmentTest
 {
   private static final ObjectMapper mapper = new TestObjectMapper();
-  private static final int TEST_VERSION = 0x7;
+  private static final int TEST_VERSION = 0x9;
```
Comment (leventov) [inline core/src/test/java/org/apache/druid/timeline/DataSegmentTest.java]: Why this change?

Comment (jihoonson) [reply core/src/test/java/org/apache/druid/timeline/DataSegmentTest.java]: This represents the segment version. Its value doesn't affect to anything, but I think it's better to use a more realistic one since version 7 no more exists.

**Comment by jihoonson** [comment]
@leventov I've just double checked and confirmed there's no missing tests.
